File: test/primes.js

let count = 1;
let max = 1000000;
let pprime = 3;
while (pprime < max) {
    let check = 3;
    let isprime = 1;
    while (check * check <= pprime) {
        if (pprime % check === 0) {
            isprime = 0;
        }
        check += 2;
    }
    count += isprime;
    pprime += 2;
}

console.log(count);

File: test/primes.lua

local count = 1
local max = 100000
local pprime = 3
while pprime < max do
    local check = 3
    local isprime = 1
    while check * check <= pprime do
        if pprime%check == 0 then
            isprime = 0
        end
        check = check + 2
    end
    count = count + isprime
    pprime = pprime + 2
end

print(count)
